src/pages/CookieClicker.vue

```diff
@@ -1,49 +1,331 @@
+  cookies.value -= building.price
+  building.price += Math.ceil(building.price * 0.15)
+  building.count++
+  checkForUpgrades()
+  checkAchievements()
 }
+
+// Total cookies per second
 let cps = computed(() => {
-    let cps = 0;
-    buildings.value.forEach(building => {
-       cps+=building.cps*building.count;
-    });
-    return cps;
-});
-setInterval(()=>{
-   cookies.value+=cps.value;
-
-   document.title ='🍪' + +cookies.value.toFixed(1) + ' Cookies!';
-},1000);
+  let total = 0
+  buildings.value.forEach(b => total += b.cps * b.count)
+  if (upgrades.value.includes('Fast Fingers')) total *= 2
+  return total
+})
+
+// Cookie generation loop
+setInterval(() => {
+  cookies.value += cps.value * (goldenCookieBoost.value ? 10 : 1)
+
+  // Update page title
+  if (goldenCookieVisible.value) {
+    document.title = '🌟 GOLDEN COOKIE! 🍪'
+  } else if (goldenCookieBoost.value) {
+    document.title = '🚀 Boosted! 🍪 ' + cookies.value.toFixed(1)
+  } else {
+    document.title = '🍪 ' + cookies.value.toFixed(1) + ' Cookies!'
+  }
+}, 1000)
+
+// Random golden cookie spawns
+onMounted(() => {
+  setInterval(() => {
+    if (!goldenCookieVisible.value && Math.random() < 0.6) {
+      spawnGoldenCookie()
+    }
+  }, 15000)
+})
+
+// Show golden cookie at random position
+function spawnGoldenCookie() {
+  goldenCookieVisible.value = true
+  goldenCookiePosition.value = {
+    top: `${Math.floor(Math.random() * 80) + 5}%`,
+    left: `${Math.floor(Math.random() * 80) + 5}%`,
+  }
+
+  goldenCookieTimeout.value = setTimeout(() => {
+    goldenCookieVisible.value = false
+  }, 5000)
+}
+
+// Handle golden cookie click
+function clickGoldenCookie() {
+  const el = document.querySelector('.golden-cookie img')
+  if (el) {
+    el.classList.add('golden-clicked')
+    setTimeout(() => el.classList.remove('golden-clicked'), 800)
+  }
+
+  goldenCookieVisible.value = false
+  goldenCookieBoost.value = true
+  show("🌟 Golden Cookie activated! x10 CPS")
+  clearTimeout(goldenCookieTimeout.value)
+
+  setTimeout(() => {
+    goldenCookieBoost.value = false
+  }, 10000)
+}
+
+// Show toast message
+function show(message) {
+  toastMessage.value = message
+  showToast.value = true
+  setTimeout(() => {
+    showToast.value = false
+  }, 3000)
+}
+
+// Unlock achievements
+function checkAchievements() {
+  const unlocked = []
 
+  if (cookies.value >= 1000) unlocked.push("Cookie Enthusiast 🍪")
+  if (buildings.value.find(b => b.name === 'Grandma')?.count >= 10) unlocked.push("Grandma's Favorite 👵")
+  if (buildings.value.find(b => b.name === 'Factory')?.count >= 1) unlocked.push("Industrialist 🏭")
+
+  unlocked.forEach(name => {
+    if (!achievements.value.includes(name)) {
+      achievements.value.push(name)
+      show("🏆 Achievement unlocked: " + name)
+    }
+  })
+}
+
+// Unlock upgrades
+function checkForUpgrades() {
+  const cursor = buildings.value.find(b => b.name === 'Cursor')
+  if (cursor.count >= 5 && !upgrades.value.includes('Fast Fingers')) {
```

Comment:
Hmm so upgrades work like achievments but they give effect? That is fine, but I wouldn't make them different thing. there is no need to add extra logic if you allready have one. Merge it with achievments. It's not specifically about being different from actual CookieClicker its more about if they share 80% logic anyway and you are gonna rewrite achievment to be more reusable anyway why not handle them togather. 
If you want you can add Buyable upgrades as well like in actual CookieClicker, but it is not needed.